scripts/ntrip_ros_base.py

```diff
@@ -0,0 +1,183 @@
+    super().__init__(name)
+    self.declare_parameters(
+      namespace='',
+      parameters=[
+        ('rtcm_frame_id', 'odom'),
+        ('nmea_max_length', NMEA_DEFAULT_MAX_LENGTH),
+        ('nmea_min_length', NMEA_DEFAULT_MIN_LENGTH),
+        ('rtcm_message_package', _MAVROS_MSGS_NAME),
+        ('reconnect_attempt_max', NTRIPBase.DEFAULT_RECONNECT_ATTEMPT_MAX),
+        ('reconnect_attempt_wait_seconds', NTRIPBase.DEFAULT_RECONNECT_ATEMPT_WAIT_SECONDS),
+      ]
+    )
+
+    # Set the log level to debug if debug is true
+    if self._debug:
+      rclpy.logging.set_logger_level(self.get_logger().name, rclpy.logging.LoggingSeverity.DEBUG)
+
+    # Read an optional Frame ID from the config
+    self._rtcm_frame_id = self.get_parameter('rtcm_frame_id').value
+
+    # Determine the type of RTCM message that will be published
+    rtcm_message_package = self.get_parameter('rtcm_message_package').value
+    if rtcm_message_package == _MAVROS_MSGS_NAME:
+      if have_mavros_msgs:
+        self._rtcm_message_type = mavros_msgs_RTCM
+        self._create_rtcm_message = self._create_mavros_msgs_rtcm_message
+      else:
+        self.get_logger().fatal('The requested RTCM package {} is a valid option, but we were unable to import it. Please make sure you have it installed'.format(rtcm_message_package))
+    elif rtcm_message_package == _RTCM_MSGS_NAME:
+      if have_rtcm_msgs:
+        self._rtcm_message_type = rtcm_msgs_RTCM
+        self._create_rtcm_message = self._create_rtcm_msgs_rtcm_message
+      else:
+        self.get_logger().fatal('The requested RTCM package {} is a valid option, but we were unable to import it. Please make sure you have it installed'.format(rtcm_message_package))
+    else:
+      self.get_logger().fatal('The RTCM package {} is not a valid option. Please choose between the following packages {}'.format(rtcm_message_package, ','.join([_MAVROS_MSGS_NAME, _RTCM_MSGS_NAME])))
+
+    # Setup the RTCM publisher
+    self._rtcm_pub = self.create_publisher(self._rtcm_message_type, 'rtcm', 10)
+
+    # Initialize the client
+    self._client = NTRIPBase(
+      logerr=self.get_logger().error,
+      logwarn=self.get_logger().warning,
+      loginfo=self.get_logger().info,
+      logdebug=self.get_logger().debug
+    )
+
+    # Get some timeout parameters for the NTRIP client
+    self._client.nmea_parser.nmea_max_length = self.get_parameter('nmea_max_length').value
+    self._client.nmea_parser.nmea_min_length = self.get_parameter('nmea_min_length').value
+    self._client.reconnect_attempt_max = self.get_parameter('reconnect_attempt_max').value
+    self._client.reconnect_attempt_wait_seconds = self.get_parameter('reconnect_attempt_wait_seconds').value
+
+  def run(self):
+    # Connect the client
+    if not self._client.connect():
+      self.get_logger().error('Unable to connect')
+      return False
+    # Setup our subscribers
+    self._nmea_sub = self.create_subscription(Sentence, 'nmea', self.subscribe_nmea, 10)
```

Comment:
Do you want to allow for using both NMEA and NavSatFix messages at the same time?  Is it a problem if both of these messages are streaming at once?  Would it make sense to have this be user configurable?

Reply:
Could also have a flag that only allows one method.  Once a NMEA sentence is received, then the NavSatFix is disabled, and vice-versa